kitty-items-js/src/workers/base-event-handler.ts

```diff
@@ -55,38 +52,39 @@ abstract class BaseEventHandler {
       // when the gap is too narrow, it returns an error.
       const blockDiff = toBlock - fromBlock;
       if (blockDiff < this.blockThreshold) {
-        console.log("skipping block, blockDiff = ", blockDiff);
+        console.log("Skipping block, blockDiff = ", blockDiff);
         continue;
       }
 
       // do our processing
       let getEventsResult, eventList;
 
       try {
-        // `getEvents` will retrieve all events of the given type within the block height range supplied.
+        // `getEventsResult` will retrieve all events of the given type within the block height range supplied.
         // See https://docs.onflow.org/core-contracts/access-api/#geteventsforheightrange
-        getEventsResult = await send([
-          getEvents(this.eventName, fromBlock, toBlock),
+        getEventsResult = await Promise.all([
+          ...this.eventNames.map((name) =>
+            send([getEvents(name, fromBlock, toBlock)])
+          ),
+        ]);
+
+        eventList = await Promise.all([
+          ...getEventsResult.map((result) => fcl.decode(result)),
         ]);
```

Comment:
```suggestion
        eventList = await Promise.all(getEventsResult.map((result) => fcl.decode(result)));
```

This should work, right?

Reply:
I think so, I'll double check, could be a 2d array somehow.